### PR TITLE
[MM-23566] deployment/terraform: add server ping

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
We hardcoded a 30 second sleep to wait server to get up. Instead of sleeping, we can ping the system endpoint and check the response to verify whether it is up or not.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-23566

### CODE REVIEW

File: deployment/terraform/create.go [hunk 1]
```diff
@@ -407,3 +410,24 @@ func (t *Terraform) getOutput() (*terraformOutput, error) {
...
+	client.HttpClient.Timeout = 10 * time.Second
+	timeout := time.After(30 * time.Second)
+
+	for {
+		select {
+		case <-timeout:
+			return fmt.Errorf("timeout after 30 seconds, server is not responding")
+		case <-time.Tick(3 * time.Second):
+			_, resp := client.GetPingWithServerStatus()
+			if resp.Header.Get("database_status") == "OK" {
```
Comment (agnivade) [inline deployment/terraform/create.go:426]: We should also check if `resp.Err != nil`. Let's get to the bottom of the filestore issue also.

File: deployment/terraform/create.go [hunk 2]
```diff
@@ -407,3 +410,26 @@ func (t *Terraform) getOutput() (*terraformOutput, error) {
...
+		case <-timeout:
+			return fmt.Errorf("timeout after 30 seconds, server is not responding")
+		case <-time.After(3 * time.Second):
+			_, resp := client.GetPingWithServerStatus()
+			if resp.Error != nil {
+				mlog.Debug("got error", mlog.Err(resp.Error))
+				mlog.Info("Waiting for the server...")
+				continue
+			}
+			mlog.Info("Server status is OK")
```
Comment (agnivade) [inline deployment/terraform/create.go:431]: I thought we were going to check for the string that comes with the response now that the file issue is taken care of ?

Comment (isacikgoz) [reply deployment/terraform/create.go:431]: We can, but it already returns an error if the server status is not OK. Basically, checking the error and string is the same 😅

Comment (agnivade) [reply deployment/terraform/create.go:431]: Resolved offline.

**Pull request merged**